state-network.md

```diff
@@ -135,9 +129,35 @@ The `StorageWitness.state_proof` MUST be for a leaf node in the account trie.  T
 
 All `Witness` objects are subject to the following validity requirements.
 
-- The nodes in a `Witness` MUST be lexically ordered by their path in the trie.
-- A `Witness` may not contain any superfluous nodes that are not needed for proving.
-- A `Witness` may not contain any superfluous nodes that could be computed from other nodes that are part of the proof.
+- A: The nodes in a `Witness` MUST be lexically ordered by their path in the trie.
+- B: The `Witness` MAY NOT contain any superfluous nodes that are not needed for proving.
+- C: The `Witness` MAY NOT contain any superfluous nodes that could be computed from other nodes that are part of the proof.
+
+###### A: Lexical Ordering
+
+The sequence of nodes in the witness must be lexically ordered by their nibbles
+path in the trie.  This results in the state root node always occuring first in
+the list of trie nodes.
+
+> This validity condition is to ensure that verifcation of the proof can be done
+in a single pass.
+
+###### B: No Extraneous Nodes
+
+A witness MAY NOT contain any nodes that are not part of the set needed to for proving.  
```

Comment:
From the context within this PR I'd say that the `MAY NOT` should be rather a `SHOULD NOT` or `MUST NOT`?